proxy.php

```diff
@@ -101,32 +102,47 @@ public function serve()
 
 		// Did we get an error when trying to fetch the image
 		$response = $this->checkRequest();
-		if (!$response) {
+		if ($response === -1)
+		{
 			// Throw a 404
 			header('HTTP/1.0 404 Not Found');
 			exit;
 		}
+		// Right, image not cached? Simply redirect, then.
+		if ($response === 0)
+		{
+			$this::redirectexit($request);
+		}
+
+		$time = $this->getTime();
```

Comment:
I don't think you need the `getTime()` function. Changing the value of the default time zone doesn't affect the result of `time()`.
```
date_default_timezone_set('America/Edmonton');
$t1 = time();
date_default_timezone_set('GMT');
$t2 = time();
print_r(array($t1, $t2));
/* Result: 
Array
(
    [0] => 1520449992
    [1] => 1520449992
)
*/
```